libraries/lib-snapping/CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+#  SPDX-License-Identifier: GPL-2.0-or-later
+
+set(TARGET lib-snapping)
+set(TARGET_ROOT ${CMAKE_CURRENT_SOURCE_DIR} )
+
+set(SOURCES
+   ProjectSnap.cpp
+   ProjectSnap.h
+   SnapUtils.cpp
+   SnapUtils.h
+)
+
+set( LIBRARIES
+   lib-project-rate
+   lib-numeric-formats
```

Comment:
As we mentioned today, maybe we join this with lib-time-frequency-selection, just to avoid too many libraries.  That did not previously have the two dependencies above, while it also depends on lib-screen-geometry and lib-project-history.  All four of these are comparatively low level things -- screen geometry is even a leaf.

Reply:
TBH the last place I would be looking for snapping would be "time-frequency-selection", but I can move the code I you want. When I have started working on this there was no `lib-time-frequency-selection` either.

Reply:
Snapping modifies what is time-selected, so it is related.

I don’t feel very strongly on this

Reply:
I do want to follow this with refactoring that demotes ProjectSelectionManager to lib-time-frequency-selection, so that will depend on lib-snapping if it still separate.

Reply:
Maybe lib-beats would push things into this registry and into the tables of numeric formats in the ideal reorganization — but it would not paint any more things green.